### PR TITLE
unstake function

### PR DESCRIPTION
**Pull request by mkhoury93**
The Unstake function is an option for the user to unstake their NFTs on either Polygon or ETH MAINNET. This happens by removing them from the staked mapping then transferring to the users' respective wallets

### CODE REVIEW

**Review by sypeer — COMMENTED** [review]
nice! can we get a test to along with this?

**Comment by mkhoury93** [comment]
> nice! can we get a test to along with this?

done

File: contracts/nft/NFTFacet.sol [hunk 1]
```diff
@@ -44,6 +44,19 @@ contract NFTFacet is RolesMods {
...
 
+    /**
+     * @notice it unstakes the user's NFTs and transfers it from the diamond
+     * back to the user
+     * @param nftIDs the nftIDs to unstake
+     */
+    function unstakeNFTs(uint256[] calldata nftIDs) external {
+        for (uint256 i; i < nftIDs.length; i++) {
+            // Unstake NFTs
+            NFTLib.unstake(nftIDs[i]);
```
Comment (passabilities) [inline contracts/nft/NFTFacet.sol:55]: we need checks to make sure these NFTs have been staked by the caller. you can do so by checking the boolean response value of the `unstake` library function like so:

```solidity
require(NFTLib.unstake(nftIDs[i]), "Teller: unstake NFT not owner");
```

also add these checks to the tests

**Review by passabilities — CHANGES_REQUESTED** [review]

File: contracts/nft/NFTFacet.sol (hunk 1 above)
Comment (mkhoury93) [reply contracts/nft/NFTFacet.sol:55]: on it

**Pull request closed without merging**